app/controllers/posts_controller.rb

```diff
@@ -41,27 +47,45 @@ def comment
       text: comment_params[:text]
     )
     if @comment.save
-      flash[:notice] = 'Comment added!'
+      redirect_to request.referrer, notice: 'Comment added!'
     else
-      flash[:error] = 'An error occured!'
+      redirect_to request.referrer, error: 'An error occured!'
     end
-    redirect_to request.referrer
+  end
+
+  def delete_comment
+    post = Post.find(params[:post_id])
+    comment = post.comments.find(params[:id])
+    authorize! :destroy, comment
+    comment.destroy
+    redirect_to request.referrer, notice: 'Your comment was successfully deleted!'
   end
 
   def like
-    @post = Post.find(params[:post_id])
-    @like = @post.likes.where(user_id: current_user.id).first
+    if current_user
+      @post = Post.find(params[:post_id])
+      @like = @post.likes.where(user_id: current_user.id).first
 
-    if @like
-      @like.destroy
-      current_user.likes.where(post: @post).destroy_all
+      if @like
+        @like.destroy
+        current_user.likes.where(post: @post).destroy_all
+      else
+        @like = current_user.likes.new(post: @post)
+        @like.save
+      end
     else
-      @like = current_user.likes.new(post: @post)
-      @like.save
+      flash[:notice] = 'You must be signed in to like a post'
     end
     redirect_to request.referrer
   end
 
```

Comment:
- [ OPTIONAL] You did good job to work on user, like, comment and post table, but I would recommend that you separate their controllers so that each would have its own file as a controller. This will make your project more well structured and easy to read your codes.